dask/array/core.py

```diff
@@ -2596,7 +2596,10 @@ def auto_chunks(chunks, shape, limit, dtype, previous_chunks=None):
 
         # How much larger or smaller the ideal chunk size is relative to what we have now
         multiplier = (
-            limit / dtype.itemsize / largest_block / np.prod(list(result.values()))
+            limit
+            / dtype.itemsize
+            / largest_block
+            / np.prod(list(r if r != 0 else 1 for r in result.values()))
```

Comment:
Maybe make that a utility function ? It's there twice.

Reply:
Sure! It's a complicated expression and if we're using it in multiple places a small utility function makes sense